pkg/gpu/safenvml/cache.go

```diff
@@ -25,15 +25,18 @@ type DeviceCache interface {
 	SMVersionSet() map[uint32]struct{}
 	// All returns all devices in the cache
 	All() []Device
+	// AllPhysicalDevices returns all root devices in the cache
+	AllPhysicalDevices() []Device
```

Comment:
consider adding also AllMigDevice to the interface to return just the mig devices